R/manage.R

```diff
@@ -0,0 +1,754 @@
+#'   used for Chrome binaries.
+#'
+#' @seealso [chrome_versions_list()]
+#'
+#' @param ... Additional path parts.
+#' @inheritParams chrome_versions_list
+#' @inheritParams with_chrome_version
+#'
+#' @return A character vector of Chrome binary paths.
+#' @name chrome_versions
+NULL
+
+#' @rdname chrome_versions
+#' @export
+chrome_versions_path_cache <- function(...) {
+  chromote_cache_path("chrome", ...)
+}
+
+# Not exported
+chromote_cache_path <- function(...) {
+  cache_base <- normalizePath(
+    tools::R_user_dir("chromote", which = "cache"),
+    mustWork = FALSE,
+    winslash = "/"
+  )
+  file.path(cache_base, ...)
+}
+
+#' @rdname chrome_versions
+#' @export
+chrome_versions_path <- function(
+  version = "latest",
+  binary = "chrome",
+  platform = NULL
+) {
+  platform <- check_platform(platform)
+  binary <- check_binary(binary)
+
+  versions <- chrome_versions_list(
+    which = "installed",
+    binary = binary,
+    platform = platform
+  )
+
+  version_og <- version
+  version <- match_version(version, versions$version)
+
+  if (is.null(version)) {
+    cli::cli_abort(
+      c(
+        "Version {.field {version_og}} of {.code {binary}} for {platform} is not installed.",
+        "i" = 'Use {.run chromote::chrome_cache_add("{version_og}", "{binary}", "{platform}")} to install, or {.run chromote::chrome_versions_list()} to list locally cached versions.'
+      )
+    )
+  }
+
+  versions[versions$version == version, ]$path
+}
+
+#' @rdname chrome_versions
+#' @export
+chrome_versions_add <- function(version, binary, platform = NULL) {
+  res <- chrome_versions_ensure(version, binary, platform)
+
+  res[["path"]]
+}
+
+#' @param ask Whether to ask before removing files.
+#'
+#' @rdname chrome_versions
+#' @export
+chrome_versions_remove <- function(
+  version,
+  binary,
+  platform = NULL,
+  ask = TRUE
+) {
+  force(version)
+  binary <- check_binary(binary, multiple = TRUE, allow_all = TRUE)
+  platform <- check_platform(platform, multiple = TRUE, allow_all = TRUE)
+
+  versions <- chrome_versions_list(
+    "installed",
+    binary = binary,
+    platform = platform
+  )
+
+  version <-
+    if (identical(version, "all")) {
+      versions$version
+    } else {
+      match_version(version, available_versions = versions$version)
+    }
+
+  # versions is already filtered by binary + platform
+  to_delete <- versions[versions$version %in% version, ]
+
+  dirs_delete <- chrome_versions_path_cache(
+    to_delete$version,
+    paste0(to_delete$binary, "-", to_delete$platform)
+  )
+
+  if (length(dirs_delete) == 0) {
+    cli::cli_inform("No cached binaries to remove.")
+    return(invisible())
+  }
+
+  if (!identical(ask, FALSE)) {
+    cli::cli_inform(
+      "Will remove {length(dirs_delete)} cached version{?s} of chrome:"
+    )
+    cli::cli_bullets(sprintf("{.path %s}", dirs_delete))
+
+    do_delete <- utils::askYesNo("Delete from cache?")
```

Comment:
I wonder if using `cli` would be better here. 

On Windows `utils::askYesNo()` is opening a new R window, and possibly asking in R console is better now. 🤷‍♂️ 

But this is not by default in `cli` - `usethis:::ui_yep` is an example of asking with `cli` package.

Reply:
![image](https://github.com/user-attachments/assets/801fb9c9-1585-4200-8a0d-3e478aec4408)

This is the window that opens right now

Reply:
yuck that's not great, I'll copy usethis.